src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/illegalidentifiername/Example2.java

```diff
@@ -0,0 +1,37 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="IllegalIdentifierName">
+      <property name="format"
+        value="(?i)^(?!(record|yield|var|permits|sealed|open|transitive|_)$).+$"/>
+    </module>
+  </module>
+</module>
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.illegalidentifiername;
+
+// xdoc section -- start
+public class Example2 {
+  public static void main(String... args) {
+    var var = 4; // violation
+    int record = 15; // violation
+    String yield = "yield"; // violation
+
+    record Record (Record record){} // 2 violations
+
+    String yieldString = "yieldString"; // ok, part of another word
+    record MyRecord(){} // ok, part of another word
```

Comment:
Don't you think formatting the comment to look like the ones in lines 32 and 34 would make the final rendering better ?

Reply:
`part of another word` I don't understand what it means. I know we had this before. Can we change it to better wording to explain reason of no violation.

Reply:
done